crates/oxc_minifier/src/compressor/mod.rs

```diff
@@ -79,6 +79,18 @@ impl<'a> Compressor<'a> {
 
     /* Statements */
 
+    /// Remove block from single line blocks
+    /// `{ block } -> block`
+    #[allow(clippy::only_used_in_recursion)] // `&self` is only used in recursion
+    fn compress_block<'b>(&self, stmt: &'b mut Statement<'a>) {
+        if let Statement::BlockStatement(block) = stmt {
+            if block.body.len() == 1 && !matches!(&block.body[0], Statement::Declaration(_)) {
```

Comment:
In C++, and I think in the context of Rust as well, reducing branches is a great way of improving performance. I feel like we should always try to reduce the number of branches, since compiler has to do lots of work on branch prediction.

Reply:
In Rust, I leave this to LLVM for the time being, and optimize it if it shows up on the profiler.

But can you teach me here, where's the extra branch? The code here needs to look at the statement and see if the inner block doesn't contain a single declaration.

Reply:
We don't want to compress
```
if (x) { var foo = 1 } 
```
to
```
if (x) var foo = 1 
```
as they lead to different semantics (due to annex b), which will break the mangle.

Reply:
I added a comment to clarify this.